src/models/four-up-grid.ts

```diff
@@ -0,0 +1,163 @@
+import { types } from "mobx-state-tree";
+
+export const DEFAULT_SPLITTER_SIZE = 3;
+
+export enum CellPositions {
+  NorthWest = 0,
+  NorthEast,
+  SouthEast,
+  SouthWest,
+}
+
+interface ICellUpdate {
+  bottom?: number;
+  height?: number;
+  left?: number;
+  right?: number;
+  top?: number;
+  width?: number;
+  scale?: number;
+}
+
+export interface IGridUpdate {
+  height?: number;
+  width?: number;
+  hSplitter?: number;
+  vSplitter?: number;
+  initSplitters?: boolean;
+  resizeSplitters?: boolean;
+}
+
+export const FourUpGridCellModel = types
+  .model("FourUpGridCell", {
+    bottom: 0,
+    height: 0,
+    left: 0,
+    right: 0,
+    scale: 0,
+    top: 0,
+    width: 0,
+  })
+  .actions((self) => {
+    const updateCell = (update: ICellUpdate) => {
+      self.bottom = typeof update.bottom !== "undefined" ? update.bottom : self.bottom;
```

Comment:
Because of the typing it has to be an exact match to "undefined" otherwise the compiler emits this error: `Type 'number | undefined' is not assignable to type 'number'.\n  Type 'undefined' is not assignable to type 'number'.`.